code/modules/reagents/chemistry/reagents/food_reagents.dm

```diff
@@ -841,6 +841,27 @@
 	taste_description = "rancid fungus"
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
 
+/datum/reagent/consumable/moltobeso
+	name = "Molt'Obeso" //pardon my Italian
+	description = "Concentrated gluttony."
+	color = "#f8fc36"
+	taste_description = "gluttony"
+	nutriment_factor = 0 //the essence of this sauce is to stimulate hunger and improve the absorption of calories from food eaten
+	metabolization_rate = 0.25 * REAGENTS_METABOLISM
+	metabolized_traits = list(TRAIT_GLUTTON)
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+
+/datum/reagent/consumable/moltobeso/on_mob_life(mob/living/carbon/affected_mob, seconds_per_tick, times_fired)
+	. = ..()
+	for(var/datum/reagent/consumable/food as anything in affected_mob.reagents.reagent_list)
+		if(!istype(food))
+			continue
```

Comment:
```suggestion
	for(var/datum/reagent/consumable/food in affected_mob.reagents.reagent_list)
```
There is no point in istyping if we're doing a loop since the loop should handle that for us

Reply:
For some reason, the loop keeps running. I tried to drink an inedible reagent, and it caused an error in the get_nutriment_factor() proc. 
![image](https://github.com/user-attachments/assets/728d317f-3986-4de3-a23a-fe14605ccf8f)

Reply:
because you kept ``as anything`` there

Reply:
a